### ISSUE TITLE
Buildings.ThermalZones.Detailed.Examples.FFD.* models fail at runtime with GC memory issues

### ISSUE THREAD

**Issue by casella** [OpenModelica/OpenModelica#10074]
All the 8 models in the `Buildings.ThermalZones.Detailed.Examples.FFD` package fail at runtime during initialization with [errors](https://libraries.openmodelica.org/branches/master/Buildings_latest/files/Buildings_latest_Buildings.ThermalZones.Detailed.Examples.FFD.RoomOnlyExteriorWallNoWindow.sim) such as:
```
GC Warning: Failed to expand heap by -4096 bytes
GC Warning: Out of Memory! Heap size: 1 MiB. Returning NULL!
src->dim_size[0] != dst->dim_size[0], 0 != -1
assert            | debug   | Failed to copy array. Dimension sizes are not equal and destination array is not flexible.
```
This should be investigated and fixed.

**Comment by casella** [comment]
Just tested it also on Dymola under Windows with Visual Studio 2022, we got a crash. Probably a binary library issue, see lbl-srg/modelica-buildings#4015

**Comment by casella** [comment]
Status update with the latest version of Buildings and OpenModelica as of 28 March:
- with the [Linux CI](https://libraries.openmodelica.org/branches/master/Buildings_latest/Buildings_latest.html)
  - ForcedConvection runs fine
  - The other fails due to memory issues
    - "GC Warning: Out of Memory! Heap size: 4 MiB. Returning NULL!" and "Failed to copy array. Dimension sizes are not equal and destination array is not flexible."
    - "free(): unaligned chunk detected in tcache 2"
- with Windows
  - ForcedConvection fails because of Windows error 0xc000005 file access violation (?)
  - NaturalConvectionWithControl fails in the same way as on the CI "Failed to copy array. Dimension sizes are not equal and destination array is not flexible."
  - MixedConvection fails because of error 0xc0000005

Simulations work fine with Dymola 2025x on Windows, using Visual Studio 2022.

Probably there is something going wrong with the binaries when using them with compilers other than VS.

@arun3688 maybe you can give a look and try to figure out what is going wrong in the interaction between OMC and the LBL-generated binaries?

**Comment by casella** [comment]
@mwetter the problem is still there, see the [Buildings_latest](https://libraries.openmodelica.org/branches/master/Buildings_latest/Buildings_latest.html) report. 

Should we investigate more on our side, or do you think you can do something on the library side?

**Comment by mwetter** [comment]
@casella That looks like it is something we need to first reproduce and isolate on the Buildings/FFD side with a minimum example. See comment on Buildings repository from just before.

**Comment by JayHuLBL** [comment]
@casella We are working on the issue with the help from the [Cline](https://cline.bot/) (an open-source autonomous AI agent extension for Visual Studio Code). Its suggestion is as below:

<img width="807" height="347" alt="Image" src="https://github.com/user-attachments/assets/798daf7b-077f-4d34-af7e-1304d09955d2" />

We did try to increase the heap size (`GC_expand_hp()`). Not sure if we are using the right function and if the heap size has indeed increased, the model still fails to run (in branch [MBL branch issue4015_ffdBinary](https://github.com/lbl-srg/modelica-buildings/tree/issue4015_ffdBinary)).

**Comment by casella** [comment]
@adrpo what do you think?

**Comment by adrpo** [comment]
The problem appears to be this:

The call has `n=0:`
`Buildings.ThermalZones.Detailed.Examples.FFD.NaturalConvectionWithControl.roo.air.cfd.assertStringsAreUnique("ports", roo.air.cfd.nPorts, {});`

but the function looks like this (I have copied this from the flat Modelica after instantiation, I have not looked in the .mo code):

```
impure function Buildings.ThermalZones.Detailed.Examples.FFD.NaturalConvectionWithControl.roo.air.cfd.assertStringsAreUnique
  input String descriptiveName "Descriptive name of what is tested, such as 'sensor' or 'ports'";
  input Integer n(min = 2) "Number of strings";  //// <<<<------ what happens in this function if we have n=0?!
  input String[n] names "Names";
  protected Boolean[n - 1] ideNam "Flag that is set to true if the name is used more than once"; //// <<<<------  this will have n-1 size!
algorithm
  if n > 1 then
    for i in 1:n - 1 loop
      for j in i + 1:n loop
        ideNam[i] := Modelica.Utilities.Strings.isEqual(names[i], names[j], true);
        if ideNam[i] then
          break;
        end if;
      end for;
    end for;
    assert(not Modelica.Math.BooleanVectors.anyTrue(ideNam), "For the CFD interface, all " + descriptiveName + " must have a name that is unique within each room.
          The following " + descriptiveName + " names are used more than once in the room model:" + Buildings.ThermalZones.Detailed.Examples.FFD.NaturalConvectionWithControl.roo.air.cfd.returnNonUniqueStrings(n, ideNam, names));
  else
    ideNam := fill(false, max(0, n - 1)); //// <<<<------ this will be fine, I guess you can use this max(0, n-1) for the size of ideNam!
  end if;
end Buildings.ThermalZones.Detailed.Examples.FFD.NaturalConvectionWithControl.roo.air.cfd.assertStringsAreUnique;
```

I also added this to the Buildings ticket: https://github.com/lbl-srg/modelica-buildings/issues/4015

**Comment by adrpo** [comment]
@perost could we actually detect these things and give a warning? 
I mean the function has: `input Integer n(min = 2)` and we give it a `n=0` as input.

**Comment by adrpo** [comment]
I guess we should instantly complain on negative dimensions. 
The annoying part is that we report this properly during runtime but with no .mo file and position information.
This will just frustrate the user more than help:  `src->dim_size[0] != dst->dim_size[0], 0 != -1`
I know the runtime so I know what to look for and I can gdb but I wonder if there is a way to give some file information on this.

**Comment by perost** [comment]
> [@perost](https://github.com/perost) could we actually detect these things and give a warning? I mean the function has: `input Integer n(min = 2)` and we give it a `n=0` as input.

That wouldn't help since the argument isn't evaluated by the frontend in this case. But I guess one easy way we could check the min/max conditions would be to generate asserts for them in the function.

Of course, in this particular case there's no point in restricting the function to `n>=2` in the first place, since rewriting it to handle any array is trivial.

**Comment by adrpo** [comment]
Yep, asserts would help and we could probably add line information as well.
